internal/common/mqtt/mqttconfig.go

```diff
@@ -120,7 +150,19 @@ func NewClient(configs ...Config) (*Client, error) {
 	copts.SetMaxReconnectInterval(1 * time.Second)
 	copts.SetOnConnectHandler(client.onConnect())
 	copts.SetConnectionLostHandler(client.onConnectionLost())
+	secure := os.Getenv("SECURE")
+	if len(secure) > 0 {
+		if strings.Compare(strings.ToLower(secure), "true") == 0 {
+			tlsconfig, err := NewTLSConfig()
+			if err != nil {
+				return nil, err
+			}
+			copts.SetTLSConfig(tlsconfig)
+			// TODO Use Username and password to provide authorization to MQTT Broker zuncomment the below two lines to enable authorization after creating password file for each user
```

Comment:
```suggestion
			// TODO: Use Username and Password to provide authorization to MQTT Broker uncomment the below two lines to enable authorization after creating password file for each user. This is temporary solution!
```

Reply:
This is not a temporary solution. Its the solution as suggested by mosquitto broker. So its user dependent and hence commented.

Reply:
1) login and password cannot be stored in the program and must be exported from a secure storage 
2) more suitable option is a signed certificate with a fixed SAN (we had difficulties with this, but I will investigate this issue in more detail later) 
So I think this is a temporary solution.

Reply:
Yes , it should be ideally but you have to pass the password through copts only thats the architecture design of Mosquitto broker . Secondly Certificates through SAN can give you secure communication channel which is already done. The issue was resolved by me. There is no issue in it. So I think there is some confusion from your side. SAN certificates were fine and are working fine now. As already mentioned in previous PR also the issue was resolved and it wasnot certificates issue.

Reply:
As I wrote above, this is a temporary solution, since the password cannot be stored in the program.

Reply:
The use of a fixed SAN certificate provides authentication without passing a password. (And while me and you have not been able to run this option)

Reply:
> As I wrote above, this is a temporary solution, since the password cannot be stored in the program.

Agree to this point of yours. But still the password has to be passed as string be it readin from any file as per the mosquitto design requirement. So i think its reading a pssword file in string ad passing a string. Now, since its user dependent, its upto the user how he wants to read it from file, or memory etc. I think this discussion can be taken in separate thread since this will not impact the current lines of code. Hence this can be approved

Reply:
FYI: a secure channel can be established with any user, including a hacker.

Reply:
> The use of a fixed SAN certificate provides authentication without passing a password. (And while me and you have not been able to run this option)

No, I was able to run with SAN certificate and password. So I have tried all three modes of security that the mosquitto broker provides and al have been running suceessfully without error

Reply:
> FYI: a secure channel can be established with any user, including a hacker.

Yes, because its role is just to provide the secure channel. Thats where the password comes into picture. Which is way of providing ACL(Acess Control List) list of users who can acess. And With SAN i have added password for twwo users and only those two users could access. 
To clear the air of Confusion following are the different modes I have tried and tested sucessfully
1. No security:: Any user can access the broker specifying the url
2. Secure Channel: Any User can access the broker providing the SAN certificates
3. Secure Channel with password :: Only specific users can access the broker after prviding the password and SAN certificates.

Reply:
Sorry. I had not reviewed this carefully. What about receiving Username and Password as parameters of REST API? I think commenting them is not a good approach.

Reply:
Let's clear situation:
**After accepting this PR, we will not be able to release, not even a v1.1.7**. Since the container downloaded from Dockerhub will not be able to work with the broker. Where in the documentation and source code will this be mentioned?

Reply:
I have commented it because its user choice that what level of security user wants. And If i uncomment it it will be always secure mode and will be checking the broker with password. Receiving username password via REST API, would make it more vulnerable I think, not sure about it,

Reply:
> Let's clear situation: **After accepting this PR, we will not be able to release, not even a v1.1.7**. Since the container downloaded from Dockerhub will not be able to work with the broker. Where in the documentation and source code will this be mentioned?

Documentation its already mentioned. SOurce code //TODO line states clearly that it has to be uncommented after generating a password file for particular user

Reply:
At a minimum, we need to put this data into an external file and edge-orchestration read them from file

Reply:
I m not denying to put password in the file and read it. Its not about how you read the password. Thats not the issue here.
I have just mentioned no matter how you read it be it API or file but in the end it has to be passed as string in copts only

Reply:
> I have commented it because its user choice that what level of security user wants. And If i uncomment it it will be always secure mode and will be checking the broker with password. Receiving username password via REST API, would make it more vulnerable I think, not sure about it,

The problem of static input is that the code is not working if there are two brokers with different password and username on AWS.

Reply:
I think how the password is read can be discussed in separate thread. Passing password thorugh copts so that broker recognises and receives it is limitation of mosquitto broker. Sepaarate PR can be raised for reading of password with better secure way after discussing with our security expert @tdrozdovsky :)
@Taras please let me know if you agree

Reply:
> > I have commented it because its user choice that what level of security user wants. And If i uncomment it it will be always secure mode and will be checking the broker with password. Receiving username password via REST API, would make it more vulnerable I think, not sure about it,
> 
> The problem of static input is that the code is not working if there are two brokers with different password and username on AWS.

The code is commented. So static input is not coming into picture. Keeping it static was purposefully to give user idea that string has to be passed. If this is creating so much confusion this commented lines can be removed altogether. But then only documentation wont give clear picture to a naive user

Reply:
@tdrozdovsky Security is too difficult... I have another question about the certificate. There is only one certificate in this PR. If brokers are different, I think they might want different certificates. Can the service applications send the cert information to edge orchestration via REST API?

Reply:
We can take it step by step
1) while we consider the option of communication with only one broker
2) we have two options: a) add the phrase that this is a temporary solution, accept this request, and urgently prepare to read the login and password from an external file; b) we finish reading the login and password from an external file or from another place

If accepted in the current version: we need remove the OpenSSF "passing" and "silver" badge from our project (because it violates security requirements ), and also the container that is downloaded from dockerhub will not work fully.

Reply:
I'm more concerned about the unprotected storage password in the code (hardcoded). And as I said earlier, we can implement the issues of secure interaction with the broker later

Reply:
We need to fix the secure location for password file storage . We can discuss that separately. And this file reading should not take much time. So once we fix the file location we can raise PR for that. @nitu-s-gupta : Please remove the lines..

Reply:
@nitu-s-gupta Please create an issue to discuss about cert and password.

Reply:
> @nitu-s-gupta Please create an issue to discuss about cert and password.

https://github.com/lf-edge/edge-home-orchestration-go/issues/471